Add user icons everywhere

jvyden420 opened the pull request:
**Whats done so far**:
- Profile Avatars
![image](https://user-images.githubusercontent.com/40577357/64580470-6fec9500-d354-11e9-8ed3-126b983dd1ed.png)
**What I want to do**:
-  ~~Avatar next to logged in status~~ Completed.
- ~~Maybe avatars in the leaderboards? Open to discussion.~~ Not happening.

Hughp135 commented on front-end/src/app/components/user-profile/user-profile.component.scss:
```diff
@@ -53,3 +53,18 @@
+    display: inline-block;
+    padding: 20px;
+    font-size: 36px;
+    position: relative; // Is this good practice?
```
I think the best practise would be to use `vertical-align` on the image, instead of re-positioning the header, but I'm not sure if it will work in that case.

I don't really mind about some CSS being not best practise, so just leave it if it works. However, can you remove these comments?

Hughp135 commented on front-end/src/app/components/user-profile/user-profile.component.scss:
```diff
@@ -53,3 +53,18 @@
+    padding: 20px;
+    font-size: 36px;
+    position: relative; // Is this good practice?
+    bottom: 10px; // I DONT KNOW! I SPENT 90 MINUTES ON THIS SO I DONT CARE, EITHER! AAAAA
```
lmao that sucks, but can we take leave this comment out now?

jvyden420 replied:
yeah i got a little salty

jvyden420 commented on front-end/src/app/components/game-lobby/user-list/user-list.component.scss:
```diff
@@ -126,3 +126,7 @@
 }
+
+.image-container img {
+  border-radius: 15%;
```
I disagree, even the osu webpage uses way more rounding than I do.
![image](https://user-images.githubusercontent.com/40577357/64650572-87c42780-d3ed-11e9-9984-36a21deaae63.png)
![image](https://user-images.githubusercontent.com/40577357/64650415-213f0980-d3ed-11e9-8045-f008c57f1409.png)

Hughp135 replied:
so the main issue i have with it is that avatars are commonly have square borders designed into them, so rounding the corners kinda cuts those off

jvyden420 replied:
i haven't seen many of these square borders as of recently, but i dont hang around in the community too often so i'll take your word for it

i still strongly disagree with this but if you insist *i guess* i can revert it

i'd still prefer to decrease the rounding a lot than to just remove it
idk pick your favorite:
![image](https://user-images.githubusercontent.com/40577357/64935969-b2651480-d821-11e9-8036-625123f1d574.png)
![image](https://user-images.githubusercontent.com/40577357/64935985-c577e480-d821-11e9-9743-60ad31b36934.png)
![image](https://user-images.githubusercontent.com/40577357/64936024-f0facf00-d821-11e9-9727-c0173d56ed62.png)

Hughp135 replied:
It's more to be consistent with the game, but I'm happy to try a slightly roudned border such as the first image if you insist

Hughp135 commented on front-end/src/app/components/user-profile/user-profile.component.scss:
```diff
@@ -59,10 +59,9 @@
   }
   img {
     border-radius: 15%;
+    vertical-align: -10px;
```
you don't use pixel values for vertical-align - see https://css-tricks.com/almanac/properties/v/vertical-align/ for examples on how to align inlign images with text

jvyden420 replied:
Ahhh, thank you. I get it now. I also fixed my alignment issues. For some strange reason, there was a margin on the h1 tag screwin' things up. Not fun.

jvyden420 commented:
New fun stuff:
![image](https://user-images.githubusercontent.com/40577357/65942754-f0eef780-e3fb-11e9-928f-b761c89a9f1b.png)
fun!!
Also, please get back to me on the comments above.
_i'm still dead set on the rounded avatars_

jvyden420 commented:
**This looks ready to merge, go ahead if you'd like.**
### PR Recap:
- Added rounding to lobby list
![image](https://user-images.githubusercontent.com/40577357/66160441-23fdda80-e5f8-11e9-8eff-43b06fc4b440.png)
- Added user's avatar to the header
![image](https://user-images.githubusercontent.com/40577357/66160530-54457900-e5f8-11e9-9df0-0d32a0d6425d.png)
- Added avatars to profiles
![image](https://user-images.githubusercontent.com/40577357/66160600-7d660980-e5f8-11e9-990c-1fa46d683fbf.png)
